### PR TITLE
NavigationView: Fix bug where items with empty MenuItemsSource would still spawn flyout

### LINKED ISSUE #3670: NavigationViewItem still shows an Empty Flyout when selected in Left Compact mode even if it has no child menu items
**Describe the bug**

Binding the NavigationViewItem's MenuItemsSource to an empty collection still shows an empty flyout when selected in Left Compact mode.

**Steps to reproduce the bug**

1. Create a NavigationView in left compact mode.
2. Bind an empty Collection to a NavigationViewItem
2. Click on the NavigationViewItem, it will show an empty flyout

**Expected behavior**

If the NavigationViewItem has no children, it should not show an empty flyout when selected in Left Compact mode

**Screenshots**

image

**Version Info**

NuGet package version: 

Microsoft.UI.Xaml 2.5.0-prerelease.201027002

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| Yes | No |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            | Yes   |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

The bug was that we only checked if MenuItemsSource is not null, but ignored it's size. This if fixed now.

## Motivation and Context

Closes #3670
## How Has This Been Tested?

Tested manually; Unfortunately, I wasn't able to find a reliable way to detect whether a flyout is open or not from an interaction test. Running an API test is not an option as we can't simulate item invokation in there.

### CODE REVIEW

File: dev/NavigationView/TestUI/Hierarchical/HierarchicalNavigationViewDataBinding.xaml.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public HierarchicalNavigationViewDataBinding()
 
             categories.Add(new Category("Menu Item 1", "MI1", "Icon", categories2, false));
             categories.Add(new Category("Menu Item 6 (Selectable)", "MI6", "Icon", categories4, true));
-            categories.Add(new Category("Menu Item 10", "MI10", "Icon", null, true));
+            categories.Add(new Category("Menu Item 10", "MI10", "Icon", new ObservableCollection<Category>(), true));
```
Comment (karenbtlai) [inline dev/NavigationView/TestUI/Hierarchical/HierarchicalNavigationViewDataBinding.xaml.cs:74]: this should follow the same format as others such as `var categories10 = new ObservableCollection<Category>()`

Comment (chingucoding) [reply dev/NavigationView/TestUI/Hierarchical/HierarchicalNavigationViewDataBinding.xaml.cs]: Fixed now.

**Comment by chingucoding** [comment]
Looks like CI stopped hearing from the test machines :/

**Comment by ranjeshj** [comment]
@chingucoding can you merge with master ?

**Review by ranjeshj — APPROVED** [review]

**Review by StephenLPeters — APPROVED** [review]
:shipit:

**Comment by StephenLPeters** [comment]
@chingucoding have you gottent to look at the test failures by chance?

**Comment by chingucoding** [comment]
Taking a look at this soon, must have missed this PR. Thanks for the reminder.

**Comment by chingucoding** [comment]
Switched to ItemsSourceView, this should work now since the issue was based on a non working cast.

**Comment by beervoley** [comment]
@chingucoding could you please fix the tests?:)

**Comment by chingucoding** [comment]
Sure I'll take a look :)

**Comment by karenbtlai** [comment]
@chingucoding is this ready to merge in?

**Comment by chingucoding** [comment]
@karenbtlai Yes this is ready to merge!

**Pull request merged**